### PR TITLE
thinkdigitalsoftware-namespace-fix

### PR DESCRIPTION
**Pull request by ThinkDigitalSoftware**
add links to github repo in pubspec.yaml
Bring namespace.dart up to pedantic standards and add types where I could 
to make it easier to understand what parameters were required.

### CODE REVIEW

File: lib/src/namespace.dart [hunk 1]
```diff
@@ -19,89 +19,78 @@ import 'package:socket_io/src/server.dart';
...
-    return fns[idx](socket, (err) {
+  //TODO: Figure out return type for this method
+  static run0(int index, List<Function> fns, Socket socket, Function fn) {
+    return fns[index](socket, (err) {
       // upon error, short-circuit
       if (err) return fn(err);
 
       // if no middleware left, summon callback
-      if (!fns[idx + 1]) return fn(null);
+      if (fns.length < index + 1) return fn(null);
```
Comment (ThinkDigitalSoftware) [inline lib/src/namespace.dart]: Please verify that this still doesn't what's intended

File: lib/src/namespace.dart [hunk 2]
```diff
@@ -166,83 +151,72 @@ class Namespace extends EventEmitter {
...
-    this.adapter.clients(this.rooms, fn);
-    this.rooms = [];
+  /// Gets a list of clients.
+  ///
+  /// @return {Namespace} self
+  /// @api public
+  ///
+  /// TODO: Fix this description or code. Add type parameters to [fn([_])]
+  ///
+  Namespace clients(fn([_])) {
```
[the remark this answers is not in the document]
Comment (jumperchen) [reply lib/src/namespace.dart]: Ok, I will review it later.

**Comment by jumperchen** [comment]
@ThinkDigitalSoftware Thanks for the contribution.

**Comment by ThinkDigitalSoftware** [comment]
You're welcome!

**Pull request merged**